e2e/e2etests/test_burn_fungible_module_asset.go

```diff
@@ -0,0 +1,96 @@
+package e2etests
+
+import (
+	"math/big"
+
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/stretchr/testify/require"
+
+	"github.com/zeta-chain/node/cmd/zetacored/config"
+	"github.com/zeta-chain/node/e2e/runner"
+	"github.com/zeta-chain/node/e2e/utils"
+	"github.com/zeta-chain/node/pkg/constant"
+	fungibletypes "github.com/zeta-chain/node/x/fungible/types"
+)
+
+func TestBurnFungibleModuleAsset(r *runner.E2ERunner, _ []string) {
+	testBurnFungibleModuleAssetZRC20(r)
+	testBurnFungibleModuleAssetZETA(r)
+}
+
+func testBurnFungibleModuleAssetZRC20(r *runner.E2ERunner) {
+	// get fungible module balance
+	balance, err := r.ETHZRC20.BalanceOf(&bind.CallOpts{}, fungibletypes.ModuleAddressEVM)
+	require.NoError(r, err)
+
+	// if the balance is zero, we need to deposit some tokens
+	if balance.Uint64() == 0 {
+		tx, err := r.ETHZRC20.Transfer(r.ZEVMAuth, fungibletypes.ModuleAddressEVM, big.NewInt(1000))
+		require.NoError(r, err)
+
+		receipt := utils.MustWaitForTxReceipt(r.Ctx, r.ZEVMClient, tx, r.Logger, r.ReceiptTimeout)
+		utils.RequireTxSuccessful(r, receipt)
+
+		balance, err = r.ETHZRC20.BalanceOf(&bind.CallOpts{}, fungibletypes.ModuleAddressEVM)
+		require.NoError(r, err)
+		require.NotZero(r, balance.Uint64())
+	}
+
+	r.Logger.Info("Sending message to burn fungible module asset")
+	msg := fungibletypes.NewMsgBurnFungibleModuleAsset(
```

Comment:
we dont specify amount in msg, so it is burning all? maybe should be specified in msg name, like burnAll or something like that, but up to you

Reply:
In practice it will only be used to burn the entire balance